router/batchrouter/handle_lifecycle.go

```diff
@@ -85,14 +85,33 @@ func (brt *Handle) Setup(
 	if config.IsSet("WORKSPACE_NAMESPACE") {
 		defaultIsolationMode = isolation.ModeWorkspace
 	}
-	isolationMode := config.GetString("BatchRouter.isolationMode", string(defaultIsolationMode))
+	isolationMode := isolation.Mode(config.GetString("BatchRouter.isolationMode", string(defaultIsolationMode)))
+	partitionFilter := func(mode isolation.Mode) func(string) bool {
+		switch mode {
+		case isolation.ModeDestination:
+			return func(destinationID string) bool {
+				brt.configSubscriberMu.RLock()
+				defer brt.configSubscriberMu.RUnlock()
+				_, ok := brt.destinationsMap[destinationID]
+				return ok
+			}
+		case isolation.ModeWorkspace:
+			return func(workspaceID string) bool {
+				brt.configSubscriberMu.RLock()
+				defer brt.configSubscriberMu.RUnlock()
+				for _, destWithSources := range brt.destinationsMap {
+					if destWithSources.Destination.WorkspaceID == workspaceID {
+						return true
+					}
+				}
```

Comment:
We are typically using workspace isolation in namespaces with a lot of customers, i.e. a lot of sources. This loop here is expensive to run during every pinger loop of every batchrouter/router.

We could maintain a workspaces set per batchrouter/router for faster filtering but still, here we will be getting false positives, e.g. we will be ending up pinging workers that don't have any jobs in jobsdb for this destination type, just because in jobsdb there are jobs for a different destination type, but for the same workspace.